metals/src/main/scala/scala/meta/internal/builds/PantsBuildTool.scala

```diff
@@ -0,0 +1,53 @@
+package scala.meta.internal.builds
+
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.pantsbuild.BloopPants
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.internal.metals.BloopInstallResult
+import scala.meta.internal.metals.Timer
+import scala.meta.internal.metals.Time
+import scala.util.Failure
+import scala.util.Success
+
+case class PantsBuildTool(userConfig: () => UserConfiguration)(
+    implicit ec: ExecutionContext
+) extends BuildTool {
+  override def toString(): String = "pants"
+  def version: String = "1.0.0"
+  def minimumVersion: String = "1.0.0"
+  def recommendedVersion: String = "1.0.0"
+  def executableName: String = "pants"
+  def digest(workspace: AbsolutePath): Option[String] = {
+    new PantsDigest(userConfig).current(workspace)
+  }
+
+  def bloopInstall(
```

Comment:
Is the plan to move some of this logic eventually to Bloop? I'm open to maintaining the appropriate code there so that we can have first-class Pants support regardless of the use of Metals like we do with the rest of the build tools.

Reply:
Once this component stabilizes I think it would be best to contribute it upstream to Pants, where it can be kept up-to-date with latest Pants changes and also maybe be improved to take advantage of more suitable Pants APIs.

So far, we're still experimenting the Pants->Bloop translation, which makes it most convenient to keep it here in Metals for now.